LibTest/io/Stdin/first_A02_t02.dart

```diff
@@ -24,17 +30,23 @@ run_process() async {
 run_main() async {
   String executable = Platform.resolvedExecutable;
   String eScript = Platform.script.toString();
-  int called = 0;
 
-  await Process.start(executable, [eScript, "test"], runInShell: true).then(
-      (Process process) async {
-    process.stdin.writeln("1");
-    await process.exitCode.then((int code) async {
-      Expect.equals(0, code);
-        called++;
-    });
-  });
-  Expect.equals(1, called);
+  Process process = await Process.start(executable, [eScript, "test"], runInShell: true);
+
+  process.stdin.writeln("1");
+  await process.stdin.flush();
+
+  try {
+    await process.stdin.close();
+    int code = await process.exitCode;
+    Expect.equals(0, code);
+  } catch (e) {
+    print("Subprocess failed or exited prematurely!");
+    print("Exit code: ${await process.exitCode}");
+    print("stdout: " + await readOutput(process.stdout));
```

Comment:
If the subprocess happens to print a lot of output (e.g. crash information, ...)  this test can timeout: After the kernel buffer of the pipe is full, the subprocess might block on a `write()`. 

Because of this situation it's always good to start draining stdout/stderr right after starting the process (and not waiting until it has quit)